### ISSUE TITLE
dosbox-x.conf should be loaded from a path relative to the doxbox-x.app bundle

### ISSUE THREAD

**Issue by jpc** [joncampbell123/dosbox-x#2243]
I would like to give a preconfigured macOS DosBox to someone else without asking them to copy files around or configure things manually.

There seems to be functionality that looks for config files next to the dosbox binary (https://github.com/joncampbell123/dosbox-x/blob/master/src/gui/sdlmain.cpp#L11539-L11549) but the `GetDOSBoxXPath` function does not seem to be implemented on macOS. This functionality (searching relative to executable path) is already available in https://github.com/joncampbell123/dosbox-x/blob/master/src/gui/sdlmain.cpp#L11407-L11424 and used to find fonts so maybe it would be a simple matter of reusing this code?

I can make a pull request but I would love to confirm that this would be a welcome change.

**Comment by jpc** [comment]
Ok, I compiled the code and digged a little bit deeper into this. The `GetDOSBoxXPath` function does work on macOS (`wai_getExecutablePath` is in fact cross-platform, the prefix made my initial heuristics see it as some WinAPI call) but returns the path to the `dosbox-x.app/Contents/MacOS` folder which does make sense. If a place my config there it does indeed pick it up. (putting custom files inside the bundle might break code signing? maybe it should?)

After figuring this out I discovered that the path given to `IMGMOUNT` seems to go straight into `fopen`, which means that it needs to be either relative to the current working dir (so `$HOME` for apps launched on macOS) or absolute. Either of these will not work well for my desired use case.

I guess changing the current working dir to `GetDOSBoxXPath` if a config file was found there would be the easiest way to solve it, but I am of two minds whether it's a hack or an elegant solution. 😅 Would it break something that I am not aware of?

**Comment by jpc** [comment]
I have made the change here, if anybody is interested:
https://github.com/jpc/dosbox-x/commit/b8948989587156d748e9c0ea3ad93782af435acc

I will test it a bit more and make a binary release later on in case anybody else finds this useful.

**Comment by emendelson** [comment]
This won't affect individual users, but Apple won't notarize applications where the `Contents/MacOS` folder contains a text file. The contents of that folder need to codesigned by the notarization tool, and this can't be done with text files. That doesn't stop you from adding a conf file to the folder after you download a notarized app, but Gatekeeper will always complain about the app with the added file.

**Comment by jpc** [comment]
@emendelson Yeah, that's what I thought. I guess the best thing to do architecturally would be to add a new bundle type and associate the main app with it. This is what Boxer already does for vanilla DosBox (and Wineskin does the same for Wine apps).
A portable DOS container format sounds nice but it would probably be a better idea to just add dosbox-x support to Boxer than to reinvent it in this codebase.

**Comment by emendelson** [comment]
@jpc You might look at what I've done here: the AppleScript is editable, and you can put in your own build (or the standard build) or do anything else you want to do with it:

http://www.columbia.edu/~em36/dosboxapp.html

As will be obvious from the AppleScript, this is based on code that I threw together over the past ten years when building other applications, and it's hopelessly inefficient and very badly written. But it seems to work reasonably well.

**Comment by jpc** [comment]
@emendelson Thanks for the link.

I tested my change (without worrying about code signing for now) and it works well so I released a (x86_64/arm64) universal binary here: https://github.com/jpc/dosbox-x/releases/tag/jpc-1 I hope it will be useful for someone.

**Comment by Wengier** [comment]
Just FYI, in the current code if you give a -defaultdir option without a parameter, then the directory of the primary config file (if any) will become the default directory for DOSBox-X. This applies to all platforms, not just macOS.
